### DISCUSSION TITLE
Control default `Windows Power Mode` that the computer switches to when plugging/unplugging

### DISCUSSION THREAD

**Discussion by Lucid90** [seerge/g-helper#3042] [Q&A] [answered]
Hello,
I have a particular device [GA402RJ] that seems to like rebooting when the `Windows Power Mode` is set to `Best Power Efficiency`.
I have made a manual Profile that the laptop can sit in while unplugged:
<img width="831" alt="{452C21D6-F47B-4940-8D21-EB250898DAB6}" src="https://github.com/user-attachments/assets/1f9f4c0b-07d5-49b7-b2d7-3057a419121d">
The issue is that when uplugging, the Power Mode goes to power efficiency and the computer reboots before  I  can make the swap to this mode.
I've confirmed this by disabling the ACPI battery driver and unplugging. The battery has no issue powering the laptop without driver. When the driver is enabled, the laptop knows the battery is used and swaps the Power Mode.
Is there a way to avoid this behavior?
Thanking ahead of time for any engagement as I'm pretty screwed by this and  can't find a fix.
Thank you,
lucid

**Comment by seerge** [comment]
@Lucid90 hello,

As I understand you are setting `Silent` mode on battery that comes paired with `Power Efficiency` power mode in windows. Just adjust the power mode setting under `Fans+Power` (like you your screenshot) to something else and it should be fine :)

But honestly, I would look into proper fix for your hardware, cause device should not be rebooting when you change power modes in windows. It rather indicates some underlying hardware issues

**Reply by Lucid90** [reply]
Yes, so regarding fixing the hardware, I've been trying that for a while now, and will probably continue to try. Long story short on this side is that I've taken it to Asus and one other  laptop repair store and both of them quoted me a mobo replacement costing as much as a new laptop, and the repair store tried to repaste my cpu/gpu and spread liquid metal on my mobo and shorted it. I've since taken it to another computer repair shop that swapped the shorted parts from a donor board and made it working as it was before but it still has this instability at low power. I was also told the PSU is fine so go figure.
In regards to the temporary mitigation, what I'd need is a way to, as you put it `adjust the power mode setting under Fans+Power`, but without immediately swapping to it, as that causes the reboot.

**Reply by seerge** [reply, marked as answer]
You can manually adjust config then. 

1. Quit G-Helper
2. Open `%appdata%\GHelper\config.json`
3. Add following line to the beginning of the file and save it

```
  "powermode_2": "00000000-0000-0000-0000-000000000000"
```

That should associate `Silent` mode in app with Balanced windows power mode

**Reply by Lucid90** [reply]
Thank you very much for your help.
I've set all the default modes to `Best Performance` *Windows Power Mode* .
<img width="524" alt="{EE461CE9-825D-41B8-81B7-7B4823928E0C}" src="https://github.com/user-attachments/assets/a67de796-af64-4d60-88f1-6bf50e82effa">
I've marked the reply as the answer. Also quick question. If we can set specific power limits on the CPU, why do we still get to set the *Windows Power Mdoe* ? I assume the power mode does something that is not implicit in only *Power Limits*. The reason I'm asking this is because I might infer what might be wrong with the motherboard when I decide to find a new computer repair shop.
Thanks again for your help btw.

**Reply by seerge** [reply]
@Lucid90 

`Power Mode` in windows defines how actively system can use CPU (it usually affects idle power consumption a lot)
`Power Limits` - just set a maximum power cap CPU is allowed to use on firmware / BIOS level (via special endpoints exposed for Armoury's Manual mode in a first place)

**Reply by staticx** [reply]
> You can manually adjust config then.
> 
> 1. Quit G-Helper
> 2. Open `%appdata%\GHelper\config.json`
> 3. Add following line to the beginning of the file and save it
> 
> ```
>   "powermode_2": "00000000-0000-0000-0000-000000000000"
> ```
> 
> That should associate `Silent` mode in app with Balanced windows power mode

How to adjust that when I change to "Turbo" it turns on the "GameTurbo/High Performance" Profile? it seems that whatever I'm choosing in Ghelper always chooses "Balanced" (in the screenshot I've chose it manually)
![Screenshot_61](https://github.com/user-attachments/assets/32de26ea-2651-42bc-90d1-c062bcc21296)

thanks!

**Reply by seerge** [reply]
@staticx you are mixing power plans and power modes. You need to change only power mode, and app does that

**Reply by staticx** [reply]
> @staticx you are mixing power plans and power modes. You need to change only power mode, and app does that

When I change power mode (in Gapp) my laptop runs slower than when I choose the "GameTurbo" plan. I'm a sound editor and on turbo mode I can hear noises in the sound, when I change manually to gameturbo plan, the noise disappear and computer runs faster.

**Reply by seerge** [reply]
@staticx if you want to use custom windows power plans, that is also possible (even tho doesn't make much sense in my opinion). Please check [readme on how to do that ](https://github.com/seerge/g-helper/wiki/Power-user-settings#custom-windows-power-plans-with-each-mode).
